### PR TITLE
Add support to use managed identities for service bus

### PR DESCRIPTION
**Pull request by shishirx34**
This adds support to `TopicClient` and `SubscriptionClient` to use managed identities.

### CODE REVIEW

File: src/NuGet.Services.ServiceBus/Constants.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Services.ServiceBus
+{
+    public class Constants
```
Comment (joelverhagen) [inline src/NuGet.Services.ServiceBus/Constants.cs:6]:
```suggestion
    internal static class Constants
```

File: src/NuGet.Services.ServiceBus/SubscriptionClientWrapper.cs [hunk 2]
```diff
@@ -15,7 +15,10 @@ public class SubscriptionClientWrapper : ISubscriptionClient
 
         public SubscriptionClientWrapper(string connectionString, string topicPath, string name, ILogger<SubscriptionClientWrapper> logger)
         {
-            _client = SubscriptionClient.CreateFromConnectionString(connectionString, topicPath, name);
+            // Use managed identity for authentication if the connection string does not specify the `SharedAccessKey`.
```
Comment (joelverhagen) [inline src/NuGet.Services.ServiceBus/SubscriptionClientWrapper.cs:18]: Could you provide an example of what the two "connection string" variants look like? perhaps in the ctor xmldocs?

**Review by joelverhagen — APPROVED** [review]

**Review by zhhyu — APPROVED** [review]

File: src/NuGet.Services.ServiceBus/SubscriptionClientWrapper.cs [hunk 3]
```diff
@@ -13,9 +13,23 @@ public class SubscriptionClientWrapper : ISubscriptionClient
...
         private readonly ILogger<SubscriptionClientWrapper> _logger;
 
+        /// <summary>
+        /// Create an instance of wrapper for `TopicClient`. Use the managed identity authentication if the `SharedAccessKey` is not
+        /// spcified in the <paramref name="connectionString"/>.
+        /// </summary>
+        /// <param name="connectionString">This can be a connection string with shared access key or a service bus endpoint URL string to be used with managed identities.
+        /// The connection string examples:
+        /// 1. Using connection string: "Endpoint=sb://nugetdev.servicebus.windows.net/;SharedAccessKeyName=<access key name>;SharedAccessKey=<access key>"
+        /// 2. Using managed identitye: "sb://nugetdev.servicebus.windows.net/"
```
Comment (joelverhagen) [inline src/NuGet.Services.ServiceBus/SubscriptionClientWrapper.cs:23]:
```suggestion
        /// 2. Using managed identity: "sb://nugetdev.servicebus.windows.net/"
```

File: src/NuGet.Services.ServiceBus/SubscriptionClientWrapper.cs [hunk 4]
```diff
@@ -13,9 +13,23 @@ public class SubscriptionClientWrapper : ISubscriptionClient
         private readonly SubscriptionClient _client;
         private readonly ILogger<SubscriptionClientWrapper> _logger;
 
+        /// <summary>
+        /// Create an instance of wrapper for `TopicClient`. Use the managed identity authentication if the `SharedAccessKey` is not
+        /// spcified in the <paramref name="connectionString"/>.
+        /// </summary>
+        /// <param name="connectionString">This can be a connection string with shared access key or a service bus endpoint URL string to be used with managed identities.
+        /// The connection string examples:
+        /// 1. Using connection string: "Endpoint=sb://nugetdev.servicebus.windows.net/;SharedAccessKeyName=<access key name>;SharedAccessKey=<access key>"
```
Comment (joelverhagen) [inline src/NuGet.Services.ServiceBus/SubscriptionClientWrapper.cs:22]: 👌 awesome!

**Review by joelverhagen — APPROVED** [review]

**Pull request merged**